common/src/main/java/io/pravega/common/io/serialization/RevisionDataOutput.java

```diff
@@ -275,6 +260,19 @@ default void writeArray(ByteArraySegment segment) throws IOException {
      */
     void writeArray(byte[] array, int offset, int length) throws IOException;
 
+    /**
+     * Serializes the given {@link BufferView}. It first writes a Compact Integer representing {@link BufferView#getLength()},
+     * followed by {@link BufferView#getLength()} bytes representing the data from the {@link BufferView}.
+     *
+     * This can be read back using {@link RevisionDataInput#readArray()} as this method serializes the data in the same
+     * ways as {@link #writeArray}.
+     *
+     * @param buffer The {@link BufferView} to serialize. Can be null (in which case an Empty array will be deserialized
+     *               by {@link RevisionDataInput#readArray})).
+     * @throws IOException If an IO Exception occurred.
+     */
+    void writeBuffer(BufferView buffer) throws IOException;
```

Comment:
We previously had a default implementation of `writeArray`, but `writeBuffer` doesn't. What's the reason for superseding and not providing a default implementation?

Reply:
`writeArray(ArrayView)` was straightforwad to implement (invoke `writeArray(segment.array(), segment.arrayOffset(), segment.getLength())`.

However `writeBuffer(BufferView)` is not, because `BufferView` may not wrap an array. It may wrap direct memory or multiple buffers, so the implementation may not be that simple.

Due to their striking similarities, I have decided to consolidate `writeArray(ArrayView)` and the new `writeBuffer(BufferView)` into the latter, since `ArrayView` extends `BufferView`. I didn't want to have 2 methods that are very much the same.

Also, moving the implementation into the actual class made testing easier. I don't think the `writeArray(ArrayView)` had any explicit tests - it was tested indirectly in the (only) implementing class' unit tests.